lib/src/utf8.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library utf8;
+
+import 'dart:convert';
+import 'dart:ffi';
+import 'dart:typed_data';
+
+/// [Utf8] implements conversion between Dart strings and null-termianted
+/// Utf8-encoded "char*" strings in C.
+class Utf8 extends Struct<Utf8> {
+  static String fromUtf8(Pointer<Utf8> str) {
+    final Pointer<Uint8> array = str.cast();
+    int count = 0x1000;
```

Comment:
This is really hackisch, please use the normal loads and stores. They should be optimized within a few days.

Reply:
I'm not as optimistic. Even if they land today, they still won't be available to Flutter/Dart stable users, who might form a negative opinion of the code. For example https://github.com/dart-lang/sdk/issues/34452#issuecomment-485255442

Either way, it's only an implementation detail of the library.

Reply:
Consider opening a tracking issue in this repo to remind us

Reply:
To avoid the weird doubling logic you can use the largest length possible:

```c++
const kMaxSmi64 = (1 << 62) - 1;
const kMaxSmi32 = (1 << 30) - 1;
sizeof<ffi.IntPtr>() == 8 ? kMaxSmi64 : kMaxSmi32;
```

Reply:
Done.